fiftyone/utils/cvat.py

```diff
@@ -4651,14 +4735,22 @@ def __init__(
         metadata,
         index=None,
         immutable_attrs=None,
+        occluded_widgets=None,
     ):
+        self.is_occluded = label_dict["occluded"]
         super().__init__(
             label_dict, class_map, attr_id_map, attributes=immutable_attrs
         )
 
         self.frame_size = (metadata["width"], metadata["height"])
         self.points = label_dict["points"]
         self.index = index
+        self.occluded_widgets = occluded_widgets or {}
+
+    def _set_occluded_attribute(self, label):
```

Comment:
Can you move this parsing to the constructor?

When I started working on this code I was confused by the various different places where attributes vs FO attribute attribute stuff was appearing in this class hierarchies methods? I updated so that all parsing is done in the constructor and then you have these clean `attributes` and `fo_attributes` dicts that already contain the formatted attributes that can then be applied via a single call to `_set_attributes()`